oozie-to-airflow/mappers/spark_mapper.py

```diff
@@ -45,190 +59,171 @@ def __init__(
         will attempt to replace the EL (if it exists) with the corresponding
         variable. If no EL var is found, it just returns text. However, if the
         tag is not found under oozie_node, then return default. If there are
         more than one with the specified tag, it uses the first one found.
         """
-        if params is None:
-            params = {}
         var = xml_utils.find_nodes_by_tag(root, tag)
 
         if var:
             # Only check the first one
-            return el_utils.replace_el_with_var(var[0].text, params=params, quote=quote)
+            return el_utils.replace_el_with_var(var[0].text, params=params, quote=False)
         return default
 
     @staticmethod
-    def parse_spark_config(config_node: ET.Element) -> Dict[str, str]:
+    def _parse_config_node(config_node: ET.Element) -> Dict[str, str]:
         conf_dict = {}
         for prop in config_node:
-            name_node = prop.find("name")
-            value_node = prop.find("value")
+            name_node = prop.find(SPARK_TAG_NAME)
+            value_node = prop.find(SPARK_TAG_VALUE)
             if name_node is not None and name_node.text and value_node is not None and value_node.text:
                 conf_dict[name_node.text] = value_node.text
         return conf_dict
 
-    def update_class_spark_opts(self, spark_opts_node: ET.Element):
+    @staticmethod
+    def _parse_spark_opts(spark_opts_node: ET.Element):
         """
         Some examples of the spark-opts element:
-
-        '--conf key=value'
-        '--conf key1=value1 value2'
-        '--conf key1="value1 value2"'
-        '--conf key1=value1 key2="value2 value3"'
-        '--conf key=value --verbose --properties-file user.properties'
+        --conf key1=value
+        --conf key2="value1 value2"
         """
+        conf = {}
         if spark_opts_node.text:
             spark_opts = spark_opts_node.text.split("--")[1:]
         else:
-            raise Exception("Spark opts node has no text: {}".format(spark_opts_node))
+            raise ParseException("Spark opts node has no text: {}".format(spark_opts_node))
         clean_opts = [opt.strip() for opt in spark_opts]
         clean_opts_split = [opt.split(maxsplit=1) for opt in clean_opts]
 
-        if ["verbose"] in clean_opts_split:
-            self.__dict__["verbose"] = True
-            clean_opts_split.remove(["verbose"])
-
         for spark_opt in clean_opts_split:
             # Can have multiple "--conf" in spark_opts
             if spark_opt[0] == "conf":
                 # Splits key1=value1 into [key1, value1]
-                conf_val = spark_opt[1].split("=", maxsplit=1)
-                self.conf[conf_val[0]] = conf_val[1]
-            else:
-                self.__dict__[spark_opt[0]] = "'" + " ".join(spark_opt[1:]) + "'"
+                conf_pairs = spark_opt[1].split("=", 1)
+                # Value is required
+                if len(conf_pairs) != 2:
+                    continue
```

Comment:
Should we fail here? I think this is one of the semantic checks that we should not silently ignore.

Reply:
I updated a PR. Thanks.